lib/app/services/api.dart

```diff
@@ -44,6 +48,67 @@ class ApiService {
     }).toList();
   }
 
+  Future<List<Course>> listCourses() async {
+    var response = await network.get('courses?');
+
+    var body = json.decode(response.body);
+
+    return Future.wait((body['data'] as List<dynamic>).map((data) async {
+      data = data as Map<String, dynamic>;
+      return Course(
+          id: Id<Course>(data['_id']),
+          name: data['name'],
+          description: data['description'],
+          teachers: [
+            for (String id in data['teacherIds']) await getUser(Id<User>(id))
+          ],
+          color: Color(int.parse('ff' + data['color'].toString().substring(1),
+              radix: 16)));
+    }).toList());
```

Comment:
The `Future.wait` method accepts an `Iterable`, so we don't even need to cast it to a list.